sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -609,4 +604,14 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
           resolved.namespace.map(quoteIfNeeded).mkString("."))
     }
   }
+
+  private object DatabaseNameInSessionCatalog {
+    def unapply(resolved: ResolvedDBObjectName): Option[String] = resolved match {
+      case ResolvedDBObjectName(catalog, _) if !isSessionCatalog(catalog) => None
+      case ResolvedDBObjectName(_, Seq(dbName)) => Some(dbName)
+      case _ =>
+        assert(resolved.nameParts.length > 1)
```

Comment:
The logic of this function is very similar to `DatabaseInSessionCatalog`. Shall we throw a similar exception if the nameParts is empty?

Reply:
BTW ResolvedNamespace and ResolvedDBObjectName are so alike..I didn't follow the related changes, but can we merge both of them into one? cc @cloud-fan

Reply:
`ResolvedDBObjectName` does not require the db object to exist, while `ResolvedNamespace` requires the namespace to exist.